ensime_shared/ensime.py

```diff
@@ -300,3 +362,13 @@ def fun_en_complete_func(self, client, findstart_and_base, base=None):
             # Invoked by vim
             findstart = findstart_and_base
         return client.complete_func(findstart, base)
+
+    def watchdog_client_server_connection(self, client):
```

Comment:
Dunno if it should support this case, but I tried killing the server processes and the client goes to a bad state just logging websocket not connected continuously. I have to restart vim session.
But this is related to my PR where I have to make some changes so recover from a disconnection is possible.